.github/workflows/lint-test.yml

```diff
@@ -33,144 +38,228 @@ jobs:
+      - name: Run sphinx-build
+        run: uv run noxfile.py --force-python ${{ steps.setup-env.outputs.python-version }} -s docs -- --keep-going -W -w $GITHUB_STEP_SUMMARY
 
   pyright:
     runs-on: ubuntu-latest
-    needs: lock-dependencies
+    needs:
+      - lock-dependencies
+      - determine-python-versions
+      - lint
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        python-version: ${{ fromJson(needs.determine-python-versions.outputs.test-python-versions) }}
         experimental: [false]
       fail-fast: false
     continue-on-error: ${{ matrix.experimental }}
 
     steps:
-    - uses: actions/checkout@v4
-
-    - name: Set up environment
-      id: setup-env
-      uses: ./.github/actions/setup-env
-      env:
-        PDM_USE_VENV: true
-        PDM_VENV_IN_PROJECT: true
-      with:
-        python-version: ${{ matrix.python-version }}
-
-    - name: Install dependencies
-      run: pdm install -d -Gspeed -Gdocs -Gvoice
-
-    - name: Add .venv/bin to PATH
-      run: dirname "$(pdm info --python)" >> $GITHUB_PATH
-
-    - name: Set pyright version
-      run: |
-        PYRIGHT_VERSION="$(pdm run python -c 'import pyright; print(pyright.__pyright_version__)')"
-        echo "PYRIGHT_VERSION=$PYRIGHT_VERSION" >> $GITHUB_ENV
-
-    - name: Run pyright (Linux)
-      uses: jakebailey/pyright-action@v2.2.1
-      id: pyright-linux
-      with:
-        version: ${{ env.PYRIGHT_VERSION }}
-        python-version: ${{ steps.setup-env.outputs.python-version }}
-        python-platform: "Linux"
-        annotate: ${{ matrix.python-version == '3.8' }}  # only add comments for one version
-        warnings: true
-
-    - name: Run pyright (Windows)
-      uses: jakebailey/pyright-action@v2.2.1
-      if: always() && (steps.pyright-linux.outcome == 'success' || steps.pyright-linux.outcome == 'failure')
-      with:
-        version: ${{ env.PYRIGHT_VERSION }}
-        python-version: ${{ steps.setup-env.outputs.python-version }}
-        python-platform: "Windows"
-        annotate: false  # only add comments for one platform (see above)
-        warnings: true
+      - uses: actions/checkout@v4
+
+      - name: Set up environment
+        id: setup-env
+        uses: ./.github/actions/setup-env
+        with:
+          python-version: ${{ matrix.python-version }}
+          uv-version: ${{ env.USE_UV_VERSION }}
+
+      - name: Install dependencies
+        run: uv run noxfile.py -s pyright --no-venv --install-only --force-python ${{ steps.setup-env.outputs.python-version }}
+
+      - name: Add .venv/bin to PATH
+        run: dirname "$(uv python find)" >> $GITHUB_PATH
+
+      - name: Set pyright version
+        id: pyright-version
+        run: |
+          PYRIGHT_VERSION="$(uv run python -c 'import pyright; print(pyright.__pyright_version__)')"
+          echo "pyright_version=$PYRIGHT_VERSION" >> $GITHUB_OUTPUT
```

Comment:
can we somehow extract the installed pyright version more easily with uv, rather than invoking a python interpreter?